src/Model.php

```diff
@@ -960,49 +975,21 @@ public function _unset(string $name)
      */
     public function addCondition($field, $operator = null, $value = null)
     {
-        if (is_array($field)) {
-            $this->conditions[] = [$field];
-
-            return $this;
-            /*
-            $or = $this->persistence->orExpr();
-
-            foreach ($field as list($field, $operator, $value)) {
-                if (is_string($field)) {
-                    $f = $this->getField($field);
-                } elseif ($field instanceof Field) {
-                    $f = $field;
-                }
-
-                $or->where($f, $operator, $value);
-            }
-
-            return $this;
-            */
-        }
-
-        if (is_string($field)) {
-            $f = $this->getField($field);
-        } else {
-            $f = $field;
-        }
-
-        if ($f instanceof Field) {
-            if ($operator === '=' || func_num_args() === 2) {
-                $v = ($operator === '=' ? $value : $operator);
-
-                if (!is_object($v) && !is_array($v)) {
-                    $f->system = true;
-                    $f->default = $v;
-                }
-            }
-        }
-
-        $this->conditions[] = func_get_args();
+        $this->scope()->add(new Model\Scope\CompoundCondition([func_get_args()]));
 
         return $this;
     }
 
+    /**
+     * Get the scope object of the Model.
+     *
+     * @return Model\Scope
```

Comment:
move type to code? (also in other placed if it makes sense)

Reply:
Done